core/modules/sql/SqlConnection.cc

```diff
@@ -126,14 +127,29 @@ SqlConnection::~SqlConnection() {
 
 bool
 SqlConnection::connectToDb(SqlErrorObject& errObj) {
-    assert(_connection.get());
-    if (_connection->connected()) return true;
+    assert(_connection != nullptr);
+    static int lastChecked = 0;
+    time_t currentTime = time(0);
```

Comment:
lastChecked and currentTime don't seem to serve a purpose. I think they can be deleted.